//test exception 6
//ackermann recursive calls are made in the wrong order 

int ackermann(int m, int n) {
    assert (m > 0);
    assert (n > 0);
    if (n == 0) {
        return ackermann(m-1, n);
    }
    else if (n > 0) {
        return ackermann(m-1, ackermann(m,n-1));
    }
    else return n+1;
}


int main() {
    return ackermann(10, 5);
}
